### PR TITLE
Vermalisa11 patch 1 linkedlist to array

### PR DESCRIPTION
**Pull request by vermalisa11**
U19CS071

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -1,12 +1,14 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-
+#include "stack.c"
```
Comment (murtraja) [inline main.c:4]: Not very happy with this.

File: main.c [hunk 2]
```diff
@@ -64,10 +70,11 @@ void problemTwoCheckPalindrome() {
...
 void problemThreeSort() {
-    int MAX = 255;
-    Stack counts[MAX];
-    for(int i=0; i<MAX; ++i) {
-        counts[i].top = NULL;
+    int max = 255;
+    
+    Stack counts[max];
+    for(int i=0; i<max; ++i) {
+        counts[i].top = -1;
```
Comment (murtraja) [inline main.c:77]: Why not call init() here as well?

File: stack.c [hunk 3]
```diff
@@ -3,57 +3,60 @@
 
 #include "stackconfig.h"
 #include "stack.h"
+void init(Stack* stack) {
+    stack->top = -1;
+}
+void push(Stack* stack, stackType input) {
+
+    if (isFull(stack)) {
```
Comment (murtraja) [inline stack.c:11]: Code reuse by calling isFull :+1:

File: stack.c [hunk 4]
```diff
@@ -3,57 +3,60 @@
...
-    newNode->data = data;
-    newNode->next = stack->top;
-    stack->top = newNode;
 }
+    stackType pop(Stack * stack) {
+        if (isEmpty(stack)) {
+            printf("Stack underflow while pop\n");
+            return -1;
+        }
+        stackType a = stack->data[stack->top];
```
Comment (murtraja) [inline stack.c:29]: avoid unnecessary changes like changing sensible variables names (data) to non-sensical ones (a)

File: stack.c [hunk 5]
```diff
@@ -3,57 +3,60 @@
...
-int isEmpty(Stack* stack) {
-    return stack->top == NULL;
-}
+    int isFull(Stack * stack) {
 
-int isFull(Stack* stack) {
-    Node* newNode = malloc(sizeof(Node));
-    if(newNode == NULL) {
-        return 1;
+        if (stack->top == 99) {
```
Comment (murtraja) [inline stack.c:49]: Avoid magic numbers like 99 in your code. Use a macro or a variable

**Comment by murtraja** [comment]
Great job! Keep it up.

File: stack.c (hunk 5 above)
Comment (vermalisa11) [reply stack.c:49]: it can be replaced by MAX-1
